packages/components/nodes/vectorstores/Postgres/Postgres_Exisiting_Multi.ts

```diff
@@ -0,0 +1,222 @@
+                rows: 6,
+                additionalParams: true,
+                optional: true
+            }
+        ]
+        this.outputs = [
+            {
+                label: 'Postgres Retriever',
+                name: 'retriever',
+                baseClasses: this.baseClasses
+            },
+            {
+                label: 'Postgres Vector Store',
+                name: 'vectorStore',
+                baseClasses: [this.type, ...getBaseClasses(TypeORMVectorStore)]
+            }
+        ]
+    }
+
+    async init(nodeData: INodeData, _: string, options: ICommonObject): Promise<any> {
+        const credentialData = await getCredentialData(nodeData.credential ?? '', options)
+        const user = getCredentialParam('user', credentialData, nodeData)
+        const password = getCredentialParam('password', credentialData, nodeData)
+        const embeddings = nodeData.inputs?.embeddings as Embeddings
+        const additionalConfig = nodeData.inputs?.additionalConfig as string
+        const output = nodeData.outputs?.output as string
+        const k = 4
+        const full_query = nodeData.inputs?.fullQuery as string
+        let additionalConfiguration = {}
+        if (additionalConfig) {
+            try {
+                additionalConfiguration = typeof additionalConfig === 'object' ? additionalConfig : JSON.parse(additionalConfig)
+            } catch (exception) {
+                throw new Error('Invalid JSON in the Additional Configuration: ' + exception)
+            }
+        }
+        // Parse filter field to json to use it for metadata filtering
+        /* let filter = {}
+        if (filter_field) {
+            try {
+                filter = typeof filter_field === 'object' ? filter_field : JSON.parse(filter_field)
+            } catch (exception) {
+                throw new Error('Invalid JSON in the Additional Configuration: ' + exception)
+            }
+        }*/
+
+        const postgresConnectionOptions = {
+            ...additionalConfiguration,
+            type: 'postgres',
+            host: nodeData.inputs?.host as string,
+            port: nodeData.inputs?.port as number,
+            username: user,
+            password: password,
+            database: nodeData.inputs?.database as string,
+            ssl: {
+                // Set sslmode to require for a secure connection
+                rejectUnauthorized: false,
+                sslmode: 'no-verify'
+            }
+        }
+
+        const args = {
+            postgresConnectionOptions: postgresConnectionOptions as DataSourceOptions
+        }
+
+        const vectorStore = await TypeORMVectorStore.fromDataSource(embeddings, args)
+
+        // query: { [key: string]: number[] } its a dictionary where the key is the field name and the value is the embedding
+        vectorStore.similarityMultiSearchWithScore = async (query: any, k: number, filter?: any, _callbacks = undefined) => {
```

Comment:
main changes are here, we modify this function to get the json object, the embeddings and to use the sql query